crates/client/src/lib.rs

```diff
@@ -61,30 +69,40 @@ where
     content: C,
     namespace_map: N,
     api: api::Client,
+    #[cfg(feature = "cli-interactive")]
```

Comment:
In general you're better off minimizing `#[cfg(feature ...)]` branches as much as possible, even if it means leaving (nominally) dead code around - these branches can end up being pretty fragile, especially if they aren't carefully tested (which itself isn't all that straightforward with rust/cargo).

As an overall strategy here I'd recommend leaning on the `disable_interactive` flag: force it to `true` when this feature is disabled and let that drive (most of) the rest of the conditional logic.

Reply:
Revising

Reply:
Fixed